client/main.js

```diff
@@ -390,18 +390,40 @@ $(document).ready(function() {
   }
 
   $('#nav-wiki-btn').on('click', function(event) {
-      if (!(event.ctrlKey || event.metaKey)) {
-          toggleWiki();
-      }
+    if (!(event.ctrlKey || event.metaKey)) {
+      toggleWiki();
+    }
   });
 
   $('.wiki-aside-action-collapse').on('click', collapseWiki);
 
+  function getWikiPath() {
+    var challengeType = window.common.challengeType;
+    var dashedName = window.common.dashedName;
+    var prefix = '';
+    // Get wiki prefix depending on type of challenge
+    // Types 0, 1 typically have solutions on the wiki prefixed with challenge-
+    // Type 5 typically has a solution on the wiki prefixed with algorithm-
+    if (challengeType === '0' || challengeType === '1') {
+      prefix = 'challenge-';
+      return prefix + dashedName;
+    }
+    if (challengeType === '5') {
+      prefix = 'algorithm-';
+      return prefix + dashedName;
+    }
+    return false;
+  }
+
   function showWiki() {
     if (!main.isWikiAsideLoad) {
       var lang = window.location.toString().match(/\/\w{2}\//);
       lang = (lang) ? lang[0] : '/en/';
       var wikiURL = '//freecodecamp.github.io/wiki' + lang;
+      var wikiPath = getWikiPath();
+      if (wikiPath) {
+        wikiURL = '//freecodecamp.github.io/wiki/en/' + wikiPath;
```

Comment:
@raisedadead 
Hey I hardcoded the language because as far as I could tell, most challenge articles only have an english version. 

I don't think directing them to 404s is a very good idea, so at least for now directing them to the challenge page for their language seems like a bad choice. So, if a user's language is not english, should we just direct them to the wiki homepage?

What I was thinking in making it english is that we at least present some solution. This might benefit some bilingual people whose language is set to their native language.